pkg/controller/customdomain/customdomain_controller.go

```diff
@@ -170,80 +154,93 @@ func createOrAddCustomAppsDomain(r *ReconcileCustomDomain, reqLogger logr.Logger
 			customdomainv1alpha1.CustomDomainConditionSecretNotFound,
 			customdomainv1alpha1.CustomDomainStateNotReady)
 		_ = r.statusUpdate(reqLogger, instance)
-		return err
+		return reconcile.Result{}, err
 	}
 
 	// set the secret name to be the name of the customdomain instance
 	secretName := instance.Name
 
-	// create or update secret in openshift-ingress
+	// create secret in the openshift-ingress namespace
 	ingressSecret := &corev1.Secret{}
 	err = r.client.Get(context.TODO(), types.NamespacedName{
 		Namespace: ingressNamespace,
 		Name:      secretName,
 	}, ingressSecret)
-	ingressSecret.Name = secretName
-	ingressSecret.Namespace = ingressNamespace
-	ingressSecret.Data = userSecret.Data
-	ingressSecret.Type = userSecret.Type
 	if err != nil {
-		err = r.client.Create(context.TODO(), ingressSecret)
-		if err != nil {
-			reqLogger.Error(err, "Error creating custom certificate secret")
-			return err
+		if errors.IsNotFound(err) {
```

Comment:
What happens if this error is not `IsNotFound` ? Seems we would assume it exists?

Reply:
Thanks, I've fixed it to return err in the else block.